app/src/main/java/org/oppia/app/administratorcontrols/AdministratorControlsViewModel.kt

```diff
@@ -9,22 +12,55 @@ import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.Ad
 import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.AdministratorControlsProfileViewModel
 import org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel.AdministratorControlsItemViewModel
 import org.oppia.app.fragment.FragmentScope
-import org.oppia.app.viewmodel.ObservableArrayList
+import org.oppia.app.model.DeviceSettings
+import org.oppia.app.model.ProfileId
 import org.oppia.app.viewmodel.ObservableViewModel
+import org.oppia.domain.profile.ProfileManagementController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
 import javax.inject.Inject
 
 /** [ViewModel] for [AdministratorControlsFragment]. */
 @FragmentScope
-class AdministratorControlsViewModel @Inject constructor() : ObservableViewModel() {
-  private val itemViewModelList: ObservableList<AdministratorControlsItemViewModel> = ObservableArrayList()
+class AdministratorControlsViewModel @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val logger: Logger,
+  private val profileManagementController: ProfileManagementController
+) : ObservableViewModel() {
+  private val routeToProfileListListener = activity as RouteToProfileListListener
+  private lateinit var userProfileId: ProfileId
 
-  fun processAdministratorControlsList(): ObservableList<AdministratorControlsItemViewModel> {
-    itemViewModelList.add(AdministratorControlsGeneralViewModel())
-    itemViewModelList.add(AdministratorControlsProfileViewModel())
-    itemViewModelList.add(AdministratorControlsDownloadPermissionsViewModel())
+
+  private val deviceSettingsLiveData: LiveData<DeviceSettings> by lazy {
+    Transformations.map(profileManagementController.getDeviceSettings(), ::processGetDeviceSettingsResult)
+  }
+
+  val administratorControlsLiveData: LiveData<List<AdministratorControlsItemViewModel>> by lazy {
+    Transformations.map(deviceSettingsLiveData, ::processAdministratorControlsList)
+  }
+
+  private fun processGetDeviceSettingsResult(deviceSettingsResult: AsyncResult<DeviceSettings>): DeviceSettings {
+    if (deviceSettingsResult.isFailure()) {
+      logger.e("AdministratorControlsFragmentPresenter", "Failed to retrieve profile", deviceSettingsResult.getErrorOrNull()!!)
```

Comment:
Replace `AdministratorControlsFragmentPresenter` with `AdministratorControlsFragment` we keep fragment/activity name only in TAG.

Reply:
Done